.github/workflows/integration-ext.yml

```diff
@@ -0,0 +1,101 @@
+name: Extension integration tests
+
+on:
+  pull_request:
+    paths:
+      - .github/workflows/integration-ext.yml
+      - .proxy-version
+      - go.sum
+      - jaeger/**
+      - multicluster/**
+      - test/integration/**
+      - viz/**
+
+permissions:
+  contents: read
+
+env:
+  GH_ANNOTATION: true
+  DOCKER_REGISTRY: ghcr.io/linkerd
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        component:
+          - cli-bin
+          - controller
+          - jaeger-webhook
+          - metrics-api
+          - policy-controller
+          - proxy
+          - tap
+          - web
+
+    runs-on: ubuntu-20.04
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+      - uses: ./.github/actions/docker-build
+        with:
+          docker-registry: ${{ env.DOCKER_REGISTRY }}
+          docker-target: linux-amd64
+          component: ${{ matrix.component }}
+      - name: Cache image archives
+        env:
+          ARCHIVES: /home/runner/archives
+        run: |
+          mkdir -p "$ARCHIVES"
+          docker save "$DOCKER_REGISTRY/${{ matrix.component }}:$TAG" > "$ARCHIVES/${{ matrix.component }}.tar"
+      - uses: actions/upload-artifact@3cea5372237819ed00197afe530f5a7ea3e805c8
+        with:
+          name: image-archives
+          path: /home/runner/archives
+
+  test:
+    needs: [build]
+    strategy:
+      matrix:
+        test:
+          - cluster-domain
+          - default-policy-deny
```

Comment:
Does this mean that these integration tests which test core functionality but rely on extensions to do (like viz) are only run if a file in an extension directory changes?  That's unfortunate.

Reply:
Should these tests of core functionality be in `integration.yml` instead?  And just leave the tests which specifically test extensions here?

Reply:
I'd really like to not have to build all of viz/web/etc, deploy prometheus, etc to test core functionality. I would much rather split up the install test to stop doing anything with viz.